entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryConverter.java

```diff
@@ -32,6 +32,17 @@ public static Query convertToEDSQuery(EntityQueryRequest queryRequest,
     if (attributeFilter != null) {
       queryBuilder.setFilter(attributeFilter);
     }
+
+    int limit = queryRequest.getLimit();
+    if (limit > 0) {
```

Comment:
what happens if you set without checking for the condition?

Reply:
hmm, you're right the check may not be needed since it will be converted to 0 as well in EDS query.